### DISCUSSION TITLE
Connecting with undefined

### DISCUSSION THREAD

**Discussion by weixin-tan** [prisma/prisma#16976] [Q&A]
In the Prisma doc, it is said that feeding undefined into arguments will simply be treated as if the argument did not exists. However, in this case when I try to `connect` with an undefined, if fails to execute.

<img width="1582" alt="image" src="https://user-images.githubusercontent.com/58685069/209074577-09a9e109-bdb4-42cc-a553-826febdda2ea.png">

May I know if this is intended? And if so is there a cleaner way to address this other than doing
```
const dataToUpdate = {
  where: {
    id: id,
  }, {
  data: ... // don't connect category here
  }
}

// conditionally insert the data
if (categoryId) {
  dataToUpdate.data['category'] = {
    connect: {
      id: categoryId
    }
  }
}

await prisma.course.update(dataToUpdate);
```

**Comment by nurul3101** [comment]
Hey @weixin-tan 👋 ,
This would be intended, because if `id` is undefined then it would be equivalent of an empty connect clause with no fields.
Like this:
```typescript
const dataToUpdate = {
  where: {
    id: id,
  }, {
  data: category: { connect: { } }
  }
}
```
You would need to pass atleast one field in the connect through which a record can be connected.

**Reply by weixin-tan** [reply]
Hi @nurul3101 

Thanks for the swift reply! Noted on this being intended. 

Is there a way to circumvent this then other than the way I have written
```ts
const dataToUpdate = {
  where: {
    id: id,
  }, {
  data: ... // don't connect category here
  }
}

// conditionally insert the data
if (categoryId) {
  dataToUpdate.data['category'] = {
    connect: {
      id: categoryId
    }
  }
}
```
because this feels like it leads a slightly messier code with one conditional for each optional field, and also buggier since we lose the type safety. Maybe there is some api that I missed (one that is maybe similar to `ConnectOrCreate` but `ConnectIfNotNull`)?

**Reply by LexQzim** [reply]
Here check this out 

https://github.com/prisma/prisma/issues/6186#issuecomment-850361188 

This is what helped me and made it cleaner.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
